cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -359,10 +359,19 @@ foreach(_VERSION_ENTRY ${TOOLCHAIN_VERSIONS_TXT})
     continue()
   endif()
 
-  # For debugging
-  message(STATUS "${_VARIABLE_NAME}: ${_VARIABLE_VALUE}")
+  # Allow values to be overridden to test specific versions during development
```

Comment:
Do we really want to keep this snippet? Also @kou

Reply:
It doesn't really belong in this PR, so I'm fine with removing it, but it does make it harder to verify compatibility with 3.0.0 for future changes manually or using CI (assuming there is not already a better way for overriding versions.txt).

Reply:
Which protobuf version is in 18.04? That is exercised in nightly builds.

Reply:
Huh, I thought I looked at this earlier and it was 3.12 or something, but I must have misread, because it's actually 3.0.0. So that's easy.

Reply:
Reverted here: f3b23a05

Reply:
This feature will be useful when when we use custom `ARROW_XXX_URL` environment variable to use other version.
But mixing environment variables and CMake variables confuses us. So another approach may be better.